### PR TITLE
circulation: add manuel fee

### PR DESCRIPTION
**Pull request by Garfield-fr**
Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Dependencies

ILS: [patron transaction: add manuel fee](https://github.com/rero/rero-ils/pull/3184)
NG-CORE: [Type: add date time picker](https://github.com/rero/ng-core/pull/534)

### CODE REVIEW

**Review by benerken — APPROVED** [review]

File: projects/admin/src/app/api/patron-transaction-api.service.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2021 RERO
```
Comment (zannkukai) [inline projects/admin/src/app/api/patron-transaction-api.service.ts:3]: 2019-2022 ?

File: projects/admin/src/app/circulation/patron/patron-transactions/patron-fee/patron-fee.component.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+import { DateTime } from 'luxon';
+
+@Component({
+  selector: 'admin-patron-fee',
+  templateUrl: './patron-fee.component.html',
+  styles: ['.p-inputtext { width: 100%; padding: 0 }']
+})
+export class PatronFeeComponent implements OnInit {
+
+  @Input() patronPid: string;
```
Comment (zannkukai) [inline projects/admin/src/app/circulation/patron/patron-transactions/patron-fee/patron-fee.component.ts:38]: missing docstring for both @Input()

File: projects/admin/src/app/circulation/patron/patron-transactions/patron-fee/patron-fee.component.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+  /** form fields */
+  formFields: FormlyFieldConfig[];
+
+  /** model */
+  model: FeeFormModel;
+
+  /** On submit event */
+  onSubmit: EventEmitter<any> = new EventEmitter();
+
+  constructor(
```
Comment (zannkukai) [inline projects/admin/src/app/circulation/patron/patron-transactions/patron-fee/patron-fee.component.ts:54]: missing docstring on all functions

File: projects/admin/src/app/circulation/patron/patron-transactions/patron-transactions.component.ts [hunk 4]
```diff
@@ -115,6 +117,24 @@ export class PatronTransactionsComponent implements OnInit, OnDestroy {
     }
   }
 
+  loadEngagedFees(): void {
+    this._patronTransactionService.emitPatronTransactionByPatron(this._patron.pid, undefined, 'open');
+  }
+
+  addFee(): void {
```
Comment (zannkukai) [inline projects/admin/src/app/circulation/patron/patron-transactions/patron-transactions.component.ts:124]: docstring

**Review by zannkukai — APPROVED** [review]

File: projects/admin/src/app/api/patron-transaction-api.service.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+   * @param _httpClient - HttpClient
+   */
+  constructor(private _httpClient: HttpClient) {}
+
+  /**
+   * Add fee
+   * @return Observable, array of records
+   */
+  addFee(model: FeeFormModel): Observable<Object> {
+    return this._httpClient.post('/api/patron_transactions/', model);
```
Comment (jma) [inline projects/admin/src/app/api/patron-transaction-api.service.ts:38]: Please use the `getEndpointByType` from `api.service.ts`.

File: projects/admin/src/app/circulation/patron/patron-transactions/patron-transactions.component.ts [hunk 6]
```diff
@@ -115,6 +117,25 @@ export class PatronTransactionsComponent implements OnInit, OnDestroy {
...
+    const modalRef = this._modalService.show(PatronFeeComponent, {
+        ignoreBackdropClick: true,
+        initialState: {
+          patronPid: this._patron.pid,
+          organisationPid: this._patron.organisation.pid
+        }
+      }
+    );
+    modalRef.content.onSubmit.pipe(first()).subscribe(() => {
```
Comment (jma) [inline projects/admin/src/app/circulation/patron/patron-transactions/patron-transactions.component.ts:134]: You need to unsubscribe to this observer.

**Review by jma — APPROVED** [review]

**Pull request merged**